### ISSUE TITLE
Results on WebquestionSP can't reimplement, 0.71 in Paper, but 0.608 I run

### ISSUE THREAD

**Issue by Xie-Minghui** [shijx12/TransferNet#3]
I run the code using the same random seed. (I don't change your code), the final results are as follows:
![image](https://user-images.githubusercontent.com/35483387/139423327-1c3cc77b-8dae-4cec-b2e3-3b8b9a073560.png)
1-hop accuracy is 0.732, 2-hop is 0.445, the total accuracy is 0.608. 
But the result of WebquestionSP in your paper is 0.714, which is much bigger than 0.608.
I worder how you train your model, or this is a mistake result of your paper

**Comment by shijx12** [comment]
![image](https://user-images.githubusercontent.com/11268407/141262364-2937fb06-7a2e-45e4-8668-6ed03d9df721.png)
![image](https://user-images.githubusercontent.com/11268407/141262566-8a7d1e7c-bf47-48c5-8991-b978cfadd1fa.png)

This is our checkpoint information. It seems that your loss is higher than ours. 
I do not know whether it is because some inconsistency of BERT initialization or huggingface version. Actually, we have observed unexpected performance drop with different BERT initialization.

**Comment by Xie-Minghui** [comment]
I also run the code of ComplexWebQuestion, the results are val: 0.49，test: 0.45.  The result in paper is 48.7. 
When you run you code, what the results on validation set and test set are?
Thank you

**Comment by Xie-Minghui** [comment]
> ![image](https://user-images.githubusercontent.com/11268407/141262364-2937fb06-7a2e-45e4-8668-6ed03d9df721.png) ![image](https://user-images.githubusercontent.com/11268407/141262566-8a7d1e7c-bf47-48c5-8991-b978cfadd1fa.png)
> 
> This is our checkpoint information. It seems that your loss is higher than ours. I do not know whether it is because some inconsistency of BERT initialization or huggingface version. Actually, we have observed unexpected performance drop with different BERT initialization.

How you initialization bert? I just use the same code as yours

**Comment by shijx12** [comment]
> I also run the code of ComplexWebQuestion, the results are val: 0.49，test: 0.45. The result in paper is 48.7. When you run you code, what the results on validation set and test set are? Thank you

We get 48.6 val accuracy. We report the results on validation set for CompWebQ in the paper.

**Comment by shijx12** [comment]
> How you initialization bert? I just use the same code as yours

We download the bert weights from Huggingface. However, Huggingface upgrades their API and model weights, which may cause some unexpected performance issues. We have observed it several times recently. 
We are uploading our checkpoint and will share the link to you soon.

**Comment by ShulinCao** [comment]
Please refer to https://cloud.tsinghua.edu.cn/f/786b9853c1d840578025/?dl=1 for our log and checkpoint.

**Comment by Xie-Minghui** [comment]
> Please refer to https://cloud.tsinghua.edu.cn/f/786b9853c1d840578025/?dl=1 for our log and checkpoint.

您好，请问可以上传您使用的Bert模型文件（pt和vocab.txt，config.json等文件），或者Bert版本也行？

**Comment by shijx12** [comment]
我们实验基本是2020年9到11月份做的，后来那台机器重装，我们只将代码和实验结果备份了出来，没有保留 ~/.cache 之类的。Bert 版本应该是某一个 2020 年的 Transformer 版本，我们后面尝试一下看能否找回当时的版本。

**Comment by xianghuisun** [comment]
![Screenshot from 2021-12-02 20-36-21](https://user-images.githubusercontent.com/89245998/144423465-e837fba7-cc47-409c-97cd-edea819a0ab9.png)

This is the experimental result on WebQSP(No code changes)
My question is that you said the WebQSP had 572 predicates in the paper. 
And your code is 
![Screenshot from 2021-12-02 20-44-37](https://user-images.githubusercontent.com/89245998/144424783-9186bd8a-7491-4b29-a080-c718e77ec43d.png)

but the fbwq_full KG I got from https://drive.google.com/drive/folders/1RlqGBMo45lTmWz9MUPTq-0KcjSd3ujxc is 
![Screenshot from 2021-12-02 20-46-17](https://user-images.githubusercontent.com/89245998/144425112-580b7c8b-0b2d-4fe7-a084-13b36fad77c7.png)
:
so should I experiment with relation_ids.del instead of relations.dict

**Comment by Huiopfsdfsdf** [comment]
> 

Hello! Very disturbing, is there a CWQ log?

**Comment by Huiopfsdfsdf** [comment]
> Please refer to https://cloud.tsinghua.edu.cn/f/786b9853c1d840578025/?dl=1 for our log and checkpoint.

Hello! Very disturbing, is there a CWQ log?
